### PR TITLE
RCB-527 Topics endpoint

### PR DESCRIPTION
**Pull request by cp2boston**
- Created new topics endpoint
- Created new topics example
- Created new topics unit test
- Updated README
- Updated Api

### CODE REVIEW

File: lib/Api.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ var translatedName = require("./nameTranslation");
 var tokens = require("./tokens");
 var syntax_dependencies = require("./syntax_dependencies");
 var transliteration = require("./transliteration");
+var topics = require("./topics.js");
```
Comment (guruxu) [inline lib/Api.js]: why `.js` only for this ep?

File: lib/topics.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * @param {function} callback - Callback function to be exectuted after the function to which it is passed is complete
+ */
+topics.prototype.getResults = function(parameters, userKey, protocol, serviceURL, callback) {
+
+    if (parameters.documentFile != null) {
+        parameters.loadFile(parameters.documentFile, parameters, userKey, protocol, serviceURL, "topics", callback);
+    } else {
+
+        // validate parameters
+        if (parameters.loadParams().content == null && parameters.loadParams().contentUri == null) {
```
Comment (guruxu) [inline lib/topics.js]: the standard js null/empty/white check is just `if (var)`

File: lib/Api.js (hunk 1 above)
Comment (cp2boston) [reply lib/Api.js]: Typed too fast.  It works but is unnecessary

File: lib/topics.js (hunk 2 above)
Comment (cp2boston) [reply lib/topics.js]: This is the way it was coded throughout and is not unusual.  If you'd like it to be reduced to the standard form, then it should be done in all of the endpoint checks.

Comment (guruxu) [reply lib/topics.js]: `null` is probably not enough, what happens if there's a `undefined` or `""`? also triple equals?

Comment (cp2boston) [reply lib/topics.js]: I'm not arguing.  I'm asking if you want it updated in every endpoint.

Comment (guruxu) [reply lib/topics.js]: That's the best.

Comment (cp2boston) [reply lib/topics.js]: Done

**Pull request merged**